### PR TITLE
[App-2476] Implement paypal UI

### PR DESCRIPTION
**Pull request by ricardo-frango**
**What does this PR do?**

   Added Paypal payment method

**Database changed?**

   No

**Where should the reviewer start?**

- [ ] PaypalScreen.kt
- [ ] AGPaymentScreenContentProvider.kt
- [ ] PaymentButtonsView.kt
- [ ] MainPaymentsView.kt

**How should this be manually tested?**

  Flow on how to test this or QA Tickets related to this use-case: [APP-2476](https://aptoide.atlassian.net/browse/APP-2476)

**What are the relevant tickets?**

  Tickets related to this pull-request: [APP-2476](https://aptoide.atlassian.net/browse/APP-2476)

**Code Review Checklist**

- [ ] Documentation on public interfaces
- [ ] Database changed? If yes - Migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] New Kotlin code has unit tests
- [ ] New flows in presenters unit tests
- [ ] Mappers/Validators with any kind of logic unit tests
- [ ] Functional tests pass

[APP-2476]: https://aptoide.atlassian.net/browse/APP-2476?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[APP-2476]: https://aptoide.atlassian.net/browse/APP-2476?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/paypal/PaypalScreen.kt [hunk 1]
```diff
@@ -0,0 +1,528 @@
...
+      )
+    }
+    Row(
+      modifier = Modifier
+        .clickable(onClick = onRemoveBillingAgreementClick)
+        .padding(start = 8.dp, top = 8.dp, bottom = 8.dp),
+      verticalAlignment = Alignment.CenterVertically
+    ) {
+      Text(
+        modifier = Modifier.padding(end = 8.dp),
```
Comment (pedroagostinhoaptoide) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/paypal/PaypalScreen.kt:315]: This padding is different from design. It is 16 in design

**Review by pedroagostinhoaptoide — CHANGES_REQUESTED** [review]
Also drop old commit from APP-2474

**Review by pedroagostinhoaptoide — COMMENTED** [review]
Rebase and change base branch to 2478, in order to have the default Loading view. This view can then replace the PaypalLoadingView

File: app-games/src/main/java/com/aptoide/android/aptoidegames/drawables/icons/Check.kt [hunk 2]
```diff
@@ -21,15 +22,15 @@ fun TestCheck() {
   )
 }
 
-fun getCheck(): ImageVector = ImageVector.Builder(
+fun getCheck(color: Color = primary): ImageVector = ImageVector.Builder(
```
Comment (G0retZ) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/drawables/icons/Check.kt:25]: Lets not have defaults

File: app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/PaymentButtonsView.kt [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  onBuyClickEnabled: Boolean = true,
+  onBuyClick: () -> Unit,
+  onOtherPaymentMethodsClick: () -> Unit,
+) {
+  Column(
+    modifier = modifier,
+    verticalArrangement = Arrangement.Bottom
+  ) {
+    AppGamesButton(
+      title = "Buy",
```
Comment (G0retZ) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/PaymentButtonsView.kt:46]: Hardcoded string without TODO

File: app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/paypal/PaypalScreen.kt [hunk 4]
```diff
@@ -0,0 +1,528 @@
...
+      modifier = Modifier
+        .padding(bottom = 16.dp)
+        .size(64.dp, 64.dp),
+      backgroundColor = Color.Transparent,
+      color = richOrange,
+      strokeWidth = 8.dp
+    )
+    textMessage?.let {
+      Text(
+        style = AppTheme.typography.headlineTitleText,
```
Comment (G0retZ) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/paypal/PaypalScreen.kt:468]: Lets use the new typography

**Review by G0retZ — CHANGES_REQUESTED** [review]

File: app-games/src/main/java/com/aptoide/android/aptoidegames/drawables/icons/Check.kt (hunk 2 above)
Comment (ricardo-frango) [reply app-games/src/main/java/com/aptoide/android/aptoidegames/drawables/icons/Check.kt:25]: fixed

File: app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/PaymentButtonsView.kt (hunk 3 above)
Comment (ricardo-frango) [reply app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/PaymentButtonsView.kt:46]: fixed

File: app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/paypal/PaypalScreen.kt (hunk 4 above)
Comment (ricardo-frango) [reply app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/paypal/PaypalScreen.kt:468]: removed and replaced by new loading

File: app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/paypal/PaypalScreen.kt (hunk 1 above)
Comment (ricardo-frango) [reply app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/paypal/PaypalScreen.kt:315]: fixed

**Comment by ricardo-frango** [comment]
> Rebase and change base branch to 2478, in order to have the default Loading view. This view can then replace the PaypalLoadingView

rebased

**Comment by ricardo-frango** [comment]
> Also drop old commit from APP-2474

rebased

File: app-games/src/main/java/com/aptoide/android/aptoidegames/appview/permissions/AppPermissionsView.kt
Comment (G0retZ) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/appview/permissions/AppPermissionsView.kt:1]: Lets not use the deprecated module `aptoide_ui`

File: app-games/src/main/java/com/aptoide/android/aptoidegames/drawables/icons/CheckBox.kt [hunk 5]
```diff
@@ -15,7 +15,7 @@ import androidx.compose.ui.unit.dp
 @Composable
 fun TestCheckBox() {
   Image(
-    imageVector = getCheckBox(Color.Black),
+    imageVector = getCheckBox(Color.Magenta),
```
Comment (G0retZ) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/drawables/icons/CheckBox.kt:18]: What for do we need this change? 🙂

File: app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/payment_methods/MainPaymentsView.kt
Comment (G0retZ) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/payment_methods/MainPaymentsView.kt:1]: Lets not use the deprecated module `aptoide_ui`

File: app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/paypal/PaypalScreen.kt [hunk 6]
```diff
@@ -0,0 +1,492 @@
...
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.tooling.preview.PreviewParameterProvider
+import androidx.compose.ui.unit.dp
+import androidx.navigation.NavGraphBuilder
+import androidx.navigation.NavType
+import androidx.navigation.navArgument
+import cm.aptoide.pt.aptoide_ui.animations.staticComposable
```
Comment (G0retZ) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/paypal/PaypalScreen.kt:38]: Lets not use the deprecated module `aptoide_ui`

**Review by G0retZ — CHANGES_REQUESTED** [review]

File: app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/paypal/PaypalScreen.kt (hunk 6 above)
Comment (ricardo-frango) [reply app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/paypal/PaypalScreen.kt:38]: will be changed in another ticket

File: app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/payment_methods/MainPaymentsView.kt
Comment (ricardo-frango) [reply app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/payment_methods/MainPaymentsView.kt:1]: not a change on this PR

File: app-games/src/main/java/com/aptoide/android/aptoidegames/drawables/icons/CheckBox.kt (hunk 5 above)
Comment (ricardo-frango) [reply app-games/src/main/java/com/aptoide/android/aptoidegames/drawables/icons/CheckBox.kt]: not a change on this PR

File: app-games/src/main/java/com/aptoide/android/aptoidegames/appview/permissions/AppPermissionsView.kt
Comment (ricardo-frango) [reply app-games/src/main/java/com/aptoide/android/aptoidegames/appview/permissions/AppPermissionsView.kt:1]: not a change on this PR

**Review by G0retZ — APPROVED** [review]

**Pull request merged**